server/app/services/export/CsvExporter.java

```diff
@@ -208,15 +161,52 @@ public void exportRecord(
 
   /**
    * Returns the answer retrieved by {@link ReadOnlyApplicantProgramService}. The value is derived
-   * from the raw value in applicant data, such as translating enum number to human readable text in
+   * from the raw value in applicant data, such as translating enum number to human-readable text in
    * default locale or mapping file key to download url.
    */
-  private String getValueFromAnswerMap(Column column, ImmutableMap<Path, String> answerMap) {
+  private String getValueFromAnswerMap(Column column, ImmutableMap<Path, AnswerData> answerMap) {
     Path path = column.jsonPath().orElseThrow();
     if (!answerMap.containsKey(path)) {
       return EMPTY_VALUE;
     }
-    return answerMap.get(path);
+
+    var answerData = answerMap.get(path);
+    // if it's not a checkbox question, get the answer out of AnswerData
+    if (!answerData.questionDefinition().getQuestionType().equals(QuestionType.CHECKBOX)) {
+      return answerData.scalarAnswersInDefaultLocale().get(path);
+    }
+
+    // if it is a checkbox question, use the optionAdminName in the Column to fill the column.
+    if (column.optionAdminName().isEmpty()) {
+      throw new IllegalStateException(
+          "CSV Column represents a checkbox question, but no option admin name is specified.");
+    }
+
+    if (!answerData.isAnswered()) {
+      return MultiOptionSelectionExportType.NOT_ANSWERED.toString();
+    }
+
+    ImmutableList<String> selectedList =
+        answerData
+            .applicantQuestion()
+            .createMultiSelectQuestion()
+            .getSelectedOptionAdminNames()
+            .orElse(ImmutableList.of());
+
+    ImmutableList<String> allOptionsShownInQuestionVersion =
+        answerData.applicantQuestion().createMultiSelectQuestion().getOptions().stream()
+            .map(LocalizedQuestionOption::adminName)
+            .collect(ImmutableList.toImmutableList());
+
+    if (!allOptionsShownInQuestionVersion.contains(column.optionAdminName().get())) {
```

Comment:
pls convert to lowercase before comparison

Reply:
But CSV exports always lowercase - applicant.enumhousehold_member[1].interest_in_programs.recreation & aquatics programs (all ages)=SELECTED and applicant.enumhousehold_member[1].interest_in_programs.recreation & aquatics programs (all ages)=NOT_AN_OPTION_AT_PROGRAM_VERSION. Casing doesn't affectCVS exports